lib/mayaUsd/render/vp2RenderDelegate/draw_item.h

```diff
@@ -89,21 +94,46 @@ class HdVP2DrawItem final : public HdDrawItem
 
     ~HdVP2DrawItem();
 
+    void AddRenderItem(MHWRender::MRenderItem* item, const HdGeomSubset* geomSubset = nullptr);
+
+    using RenderItemDataVector = std::vector<RenderItemData>;
+    RenderItemDataVector& GetRenderItems() { return _renderItems; }
+
     /*! \brief  Get access to render item data.
      */
-    RenderItemData& GetRenderItemData() { return _renderItemData; }
+    RenderItemData& GetRenderItemData()
+    {
+        TF_VERIFY(_renderItems.size() == 1);
+        return _renderItems[0];
+    }
+
+    /*! \brief Get acces to render item data.
+     */
+    const RenderItemData& GetRenderItemData() const
+    {
+        TF_VERIFY(_renderItems.size() == 1);
```

Comment:
I'm not sure I understand this check. We allow the addition of render items, so why verify that it's always 1?

Reply:
The function and the other similar functions that access `RenderItemData` without specifying which render item assume that the `HdVP2DrawItem` is functioning the old way where only one render item was possible. I did this to minimize the the impact of this change on the basicCurve code. The TF_VERIFY is to protect against future changes misusing these functions and using single render item access in situations where there are multiple render items per draw item.

Eventually, we need to refactor basisCurve & mesh to unify how they work and all the code to manage render item access is going to be in the parent class.